datastore.go

```diff
@@ -60,8 +62,8 @@ var DefaultOptions Options
 func init() {
 	DefaultOptions = Options{
 		GcDiscardRatio: 0.2,
-		MaxGcDuration:  1 * time.Minute,
-		GcInterval:     45 * time.Minute,
+		GcInterval:     15 * time.Minute,
+		GcSleep:        10 * time.Second,
```

Comment:
I reduced these as probabilistically sampling a log every 10 seconds shouldn't be _that_ expensive.

Rational behind this algorithm:

1. If we assume that deletes are randomly distributed, having one value log ready for garbage collection _should_ corallite with other value logs being ready.
2. After we do a full pass through all value logs, we shouldn't need to GC for a while.

That's why I have the short sleep/long sleep system.

Reply:
@Stebalien In your first point, do you mean to say that if a sample of a value log file "hits" the discard ratio, the probability that the next log file too will do so goes up ?

Can you explain this in a bit more detail ?

Reply:
Yes, but, actually, I'm not so sure about my assumption.

* Assumption: Deletes are randomly distributed between all the value logs.
* Conclusion: At any given point in time, all value logs should have approximately the same number of discarded items. Therefore, if any one value log is ready to be garbage collected, others are also likely to be ready for garbage collection.

That's not _quite_ correct. Values in a given value log are temporally correlated so deletes aren't likely to be completely random. However, the fact that enough time has passed for one value log to collect garbage is still a good indication that another value log may also have collected enough garbage for compaction.